src/main/java/com/google/sps/servlets/AbstractEventsServlet.java

```diff
@@ -0,0 +1,65 @@
+package com.google.sps.servlets;
+
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.googleapis.extensions.appengine.auth.oauth2.AppIdentityCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Abstract servlet that contains shared variables and functionality between EventsServlet and CalendarServlet.
+ */
+abstract class AbstractEventsServlet extends HttpServlet {
+  protected static final String TIMEZONE = "America/Los_Angeles";
+
+  protected static final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+  // TODO: Get the proper scopes. CalendarScopes might be outdated and not have all the proper scopes.
+  protected static final List<String> SCOPES = Lists.newArrayList(CalendarScopes.all());
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  protected String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
```

Comment:
You can use a ternary operator to simplify this function:

```
String value = request.getParameter(name);
return value == null ? defaultValue : value;
```

Ternary operators: https://www.programiz.com/java-programming/ternary-operator